lib/qmp.ml

```diff
@@ -341,9 +357,14 @@ let json_of_message = function
     in
     let args = match args with [] -> [] | args -> [ "arguments", `Assoc args ] in
     `Assoc (("execute", `String cmd) :: id @ args)
-  | Event {timestamp; event} ->
-    let secs, usecs = timestamp in
-    `Assoc [("event", `String event); ("timestamp", `Assoc [ "seconds", `Int secs; "microseconds", `Int usecs])]
+  | Event(event, data) ->
+    let secs, usecs = event.timestamp in
+    let event_data = match data with
+    | None -> `Assoc []
+    | Some x -> (match x with
+      | RTC_CHANGE r -> `Assoc [ "offset", `Int r ])
+    in
+    `Assoc [("event", `String event.event); ("timestamp", `Assoc [ "seconds", `Int secs; "microseconds", `Int usecs]); ("data", event_data)]
```

Comment:
If you append the data only when "event_data" is not `None`, as done by other fields, you would avoid changing the old `powerdown` and `block-io-error` serialisation.

Reply:
You could do it e.g. as follows
```ocaml
let event_data = match data with
| None -> []
| Some x -> begin
  let data = match x with
    | RTC_CHANGE r -> `Assoc [ "offset", `Int r ]
  in ("data", data)
end in
`Assoc (
  [ ("event", `String event.event)
  ; ("timestamp", `Assoc [ "seconds", `Int secs; "microseconds", `Int usecs])
  ] :: event_data
)
```

Reply:
Thanks, updated accordingly.
Because the Event type is: `Event of (event * event_data option)`
So we need `None` for `powerdown` and `block-io-error`

Reply:
Sure, we need to remember to bump the major version when releasing it so that we explicitly track the backward incompatible change.